ProcessLib/ThermalTwoPhaseFlowWithPP/ThermalTwoPhaseFlowWithPPMaterialProperties.cpp

```diff
@@ -0,0 +1,188 @@
+/**
+* \copyright
+* Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+*            Distributed under a Modified BSD License.
+*              See accompanying file LICENSE.txt or
+*              http://www.opengeosys.org/project/license
+*
+*/
+
+#include "ThermalTwoPhaseFlowWithPPMaterialProperties.h"
+#include <logog/include/logog.hpp>
+#include "BaseLib/reorderVector.h"
+#include "MaterialLib/Fluid/FluidProperty.h"
+#include "MaterialLib/PorousMedium/Porosity/Porosity.h"
+#include "MaterialLib/PorousMedium/Storage/Storage.h"
+#include "MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/CapillaryPressureSaturation.h"
+#include "MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/CreateCapillaryPressureModel.h"
+#include "MaterialLib/PorousMedium/UnsaturatedProperty/RelativePermeability/CreateRelativePermeabilityModel.h"
+#include "MaterialLib/PorousMedium/UnsaturatedProperty/RelativePermeability/RelativePermeability.h"
+#include "MaterialLib/TwoPhaseModels/TwoPhaseFlowWithPPMaterialProperties.h"
+#include "MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.h"
+#include "MeshLib/Mesh.h"
+#include "MeshLib/PropertyVector.h"
+#include "NumLib/NewtonRaphson.h"
+#include "ProcessLib/Parameter/Parameter.h"
+#include "ProcessLib/Parameter/SpatialPosition.h"
+
+/**
+* Regarding the details of the material propertiy functions
+* please refer to: Class H, Helmig R, Bastian P. Numerical simulation of
+* non- isothermal multiphase multicomponent processes in Porous Media�C�C1. An
+* efficient solution technique. Adv Water Resour, 2002.
+*/
+
+using MaterialLib::PhysicalConstant::MolarMass::Water;
+using MaterialLib::PhysicalConstant::MolarMass::Air;
+using MaterialLib::PhysicalConstant::IdealGasConstant;
+namespace ProcessLib
+{
+namespace ThermalTwoPhaseFlowWithPP
+{
+ThermalTwoPhaseFlowWithPPMaterialProperties::
+    ThermalTwoPhaseFlowWithPPMaterialProperties(
+        std::unique_ptr<MaterialLib::TwoPhaseFlowWithPP::
+                            TwoPhaseFlowWithPPMaterialProperties>
+            two_phase_material_model,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>
```

Comment:
--> ``std::unique_ptr<MaterialLib::Fluid::FluidProperty>&&``. The same for the types of other similar arguments.